internal/ingress/controller/config/config.go

```diff
@@ -91,7 +91,7 @@ const (
 
 // Configuration represents the content of nginx.conf file
 type Configuration struct {
-	defaults.Backend `json:",squash"`
```

Comment:
Fixing the unknown JSON option:

> internal/ingress/controller/config/config.go:94:19: SA5008: unknown JSON option "squash" (staticcheck)
	defaults.Backend `json:",squash"`

Reply:
What happens if we just remove this squash marker? :)

Reply:
See the following folding block if removing  `json:",squash"`
<details>
<summary>Testing Output</summary>
<pre>
W0629 09:47:06.690822  161331 configmap.go:452] error code 200 is not valid for custom error pages
W0629 09:47:06.690912  161331 configmap.go:452] error code 999 is not valid for custom error pages
W0629 09:47:06.691100  161331 configmap.go:343] proxy-protocol-header-timeout of 3zxs encountered an error while being parsed time: unknown unit "zxs" in duration "3zxs". Switching to use default value instead.
W0629 09:47:06.697310  161331 configmap.go:157] demo is not a valid http code: strconv.Atoi: parsing "demo": invalid syntax
W0629 09:47:06.697357  161331 configmap.go:202] 3.3.3 is not a valid textual representation of an IP address
W0629 09:47:06.697372  161331 configmap.go:202] 33:33:33::33::33 is not a valid textual representation of an IP address
W0629 09:47:06.700312  161331 configmap.go:157] demo is not a valid http code: strconv.Atoi: parsing "demo": invalid syntax
W0629 09:47:06.700344  161331 configmap.go:202] 3.3.3 is not a valid textual representation of an IP address
W0629 09:47:06.700358  161331 configmap.go:202] 33:33:33::33::33 is not a valid textual representation of an IP address
--- FAIL: TestMergeConfigMapToStruct (0.01s)
    /home/azureuser/code/ingress-nginx/internal/ingress/controller/template/configmap_test.go:117: unexpected diff: (-got +want)
         {
          Backend: {
           AppRoot: "",
           CustomHTTPErrors: [
            300,
            400,
           ],
           PreserveTrailingSlash: false,
           ProxyBodySize: "1m",
           ProxyConnectTimeout: 5,
        -  ProxyReadTimeout: 60,
        -  ProxySendTimeout: 60,
        +  ProxyReadTimeout: 1,
        +  ProxySendTimeout: 2,
           ProxyBuffersNumber: 4,
           ProxyBufferSize: "4k",
           ProxyCookiePath: "off",
           ProxyCookieDomain: "off",
           ProxyNextUpstream: "error timeout",
           ProxyNextUpstreamTimeout: 0,
           ProxyNextUpstreamTries: 3,
           ProxyRedirectFrom: "off",
           ProxyRedirectTo: "off",
           ProxyRequestBuffering: "on",
           Resolver: [
           ],
           SkipAccessLogURLs: [
            "/log",
            "/demo",
            "/test",
           ],
           SSLRedirect: true,
           ForceSSLRedirect: false,
           UsePortInRedirects: false,
           UpstreamHashBy: "",
           UpstreamHashBySubset: false,
           UpstreamHashBySubsetSize: 0,
           LoadBalancing: "",
           WhitelistSourceRange: [
           ],
           DenylistSourceRange: [
           ],
           LimitRate: 0,
           LimitRateAfter: 0,
           ProxyBuffering: "off",
           ProxyHTTPVersion: "1.1",
           ProxyMaxTempFileSize: "1024m",
           ServiceUpstream: false,
          },
          AllowSnippetAnnotations: true,
          AnnotationValueWordBlocklist: "",
          AddHeaders: "",
          AllowBackendServerHeader: false,
          AccessLogParams: "buffer=4k gzip",
          EnableAccessLogForDefaultBackend: false,
          AccessLogPath: "/var/log/test/access.log",
          HttpAccessLogPath: "",
          StreamAccessLogPath: "",
          WorkerCPUAffinity: "",
          ErrorLogPath: "/var/log/test/error.log",
          EnableModsecurity: false,
          EnableOCSP: false,
          EnableOWASPCoreRules: false,
          ModsecuritySnippet: "",
          ClientHeaderBufferSize: "1k",
          ClientHeaderTimeout: 60,
          ClientBodyBufferSize: "8k",
          ClientBodyTimeout: 60,
          DisableAccessLog: true,
          DisableHTTPAccessLog: false,
          DisableStreamAccessLog: false,
          DisableIpv6DNS: true,
          DisableIpv6: false,
          EnableUnderscoresInHeaders: false,
          IgnoreInvalidHeaders: true,
          RetryNonIdempotent: false,
          ErrorLogLevel: "notice",
          HTTP2MaxFieldSize: "",
          HTTP2MaxHeaderSize: "",
          HTTP2MaxRequests: 0,
          HTTP2MaxConcurrentStreams: 128,
          HSTS: true,
          HSTSIncludeSubdomains: true,
          HSTSMaxAge: "15724800",
          HSTSPreload: false,
          KeepAlive: 75,
          KeepAliveRequests: 1000,
          LargeClientHeaderBuffers: "4 8k",
          LogFormatEscapeNone: false,
          LogFormatEscapeJSON: false,
          LogFormatUpstream: "$remote_addr - $remote_user [$time_local] \"$request\" $status $body_bytes_sent \"$http_referer\" \"$http_user_agent\" $request_length $request_time [$proxy_upstream_name] [$proxy_alternative_upstream_name] $upstream_addr $upstream_response_length $upstream_response_time $upstream_status $req_id",
          LogFormatStream: "[$remote_addr] [$time_local] $protocol $status $bytes_sent $bytes_received $session_time",
          EnableMultiAccept: true,
          MaxWorkerConnections: 16384,
          MaxWorkerOpenFiles: 0,
          MapHashBucketSize: 64,
          NginxStatusIpv4Whitelist: [
           "127.0.0.1",
           "10.0.0.0/24",
          ],
          NginxStatusIpv6Whitelist: [
           "::1",
           "2001::/16",
          ],
          Plugins: [
          ],
          ProxyRealIPCIDR: [
           "1.1.1.1/8",
           "2.2.2.2/24",
          ],
          ProxySetHeaders: "",
          ServerNameHashMaxSize: 1024,
          ServerNameHashBucketSize: 0,
          ProxyHeadersHashMaxSize: 512,
          ProxyHeadersHashBucketSize: 64,
          ShowServerTokens: false,
          SSLCiphers: "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384",
          SSLECDHCurve: "auto",
          SSLDHParam: "",
          SSLProtocols: "TLSv1.2 TLSv1.3",
          SSLEarlyData: false,
          SSLSessionCache: true,
          SSLSessionCacheSize: "10m",
          SSLSessionTickets: false,
          SSLSessionTicketKey: "",
          SSLSessionTimeout: "10m",
          SSLBufferSize: "4k",
          SSLRejectHandshake: false,
          UseProxyProtocol: true,
          ProxyProtocolHeaderTimeout: 5000000000,
          UseGzip: false,
          UseGeoIP: true,
          UseGeoIP2: false,
          EnableBrotli: false,
          BrotliLevel: 4,
          BrotliMinLength: 20,
          BrotliTypes: "application/xml+rss application/atom+xml application/javascript application/x-javascript application/json application/rss+xml application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/svg+xml image/x-icon text/css text/javascript text/plain text/x-component",
          UseHTTP2: true,
          GzipDisable: "msie6",
          GzipLevel: 9,
          GzipMinLength: 1024,
          GzipTypes: "text/html",
          WorkerProcesses: "2",
          WorkerShutdownTimeout: "99s",
          VariablesHashBucketSize: 256,
          VariablesHashMaxSize: 2048,
          UpstreamKeepaliveConnections: 320,
          UpstreamKeepaliveTime: "1h",
          UpstreamKeepaliveTimeout: 60,
          UpstreamKeepaliveRequests: 10000,
          LimitConnZoneVariable: "$binary_remote_addr",
          ProxyStreamTimeout: "600s",
          ProxyStreamNextUpstream: true,
          ProxyStreamNextUpstreamTimeout: "600s",
          ProxyStreamNextUpstreamTries: 3,
          ProxyStreamResponses: 1,
          ProxyHTTPVersion: "",
          BindAddressIpv4: [
           "1.1.1.1",
           "2.2.2.2",
          ],
          BindAddressIpv6: [
           "[2001:db8:a0b:12f0::1]",
           "[3731:54:65fe:2::a7]",
          ],
          UseForwardedHeaders: false,
          EnableRealIp: false,
          ForwardedForHeader: "X-Forwarded-For",
          ComputeFullForwardedFor: false,
          GenerateRequestID: true,
          ProxyAddOriginalURIHeader: false,
          EnableOpentracing: false,
          OpentracingOperationName: "",
          OpentracingLocationOperationName: "",
          OpentracingTrustIncomingSpan: true,
          EnableOpentelemetry: false,
          OpentelemetryConfig: "/etc/nginx/opentelemetry.toml",
          OpentelemetryOperationName: "",
          OpentelemetryTrustIncomingSpan: true,
          OtlpCollectorHost: "",
          OtlpCollectorPort: "4317",
          OtelServiceName: "nginx",
          OtelSampler: "AlwaysOn",
          OtelSamplerRatio: 0.009999999776482582,
          OtelSamplerParentBased: true,
          OtelMaxQueueSize: 2048,
          OtelScheduleDelayMillis: 5000,
          OtelMaxExportBatchSize: 512,
          ZipkinCollectorHost: "",
          ZipkinCollectorPort: 9411,
          ZipkinServiceName: "nginx",
          ZipkinSampleRate: 1,
          JaegerCollectorHost: "",
          JaegerCollectorPort: 6831,
          JaegerEndpoint: "",
          JaegerServiceName: "nginx",
          JaegerPropagationFormat: "jaeger",
          JaegerSamplerType: "const",
          JaegerSamplerParam: "1",
          JaegerSamplerHost: "http://127.0.0.1",
          JaegerSamplerPort: 5778,
          JaegerTraceContextHeaderName: "",
          JaegerDebugHeader: "",
          JaegerBaggageHeader: "",
          JaegerTraceBaggageHeaderPrefix: "",
          DatadogCollectorHost: "",
          DatadogCollectorPort: 8126,
          DatadogEnvironment: "prod",
          DatadogServiceName: "nginx",
          DatadogOperationNameOverride: "nginx.handle",
          DatadogPrioritySampling: true,
          DatadogSampleRate: 1,
          MainSnippet: "",
          HTTPSnippet: "",
          ServerSnippet: "",
          StreamSnippet: "",
          LocationSnippet: "",
          HTTPRedirectCode: 308,
          ReusePort: true,
          HideHeaders: [
          ],
          LimitReqStatusCode: 503,
          LimitConnStatusCode: 503,
          EnableSyslog: false,
          SyslogHost: "",
          SyslogPort: 514,
          NoTLSRedirectLocations: "/.well-known/acme-challenge",
          NoAuthLocations: "/.well-known/acme-challenge",
          GlobalExternalAuth: {
           URL: "",
           Host: "",
           SigninURL: "",
           SigninURLRedirectParam: "",
           Method: "",
           ResponseHeaders: [
            "",
           ],
           RequestRedirect: "",
           AuthSnippet: "",
           AuthCacheKey: "",
           AuthCacheDuration: [
           ],
           ProxySetHeaders: {
           },
           AlwaysSetCookie: false,
          },
        - Checksum: "8573471850785068571",
        + Checksum: "12759999279689983900",
          BlockCIDRs: [
          ],
          BlockUserAgents: [
          ],
          BlockReferers: [
          ],
          LuaSharedDicts: {
           balancer_ewma: 10240,
           balancer_ewma_last_touched_at: 10240,
           balancer_ewma_locks: 1024,
           certificate_data: 20480,
           certificate_servers: 5120,
           configuration_data: 20480,
           global_throttle_cache: 10240,
           ocsp_response_cache: 5120,
          },
          DefaultSSLCertificate: nil,
          ProxySSLLocationOnly: false,
          DefaultType: "text/plain",
          GlobalRateLimitMemcachedHost: "",
          GlobalRateLimitMemcachedPort: 11211,
          GlobalRateLimitMemcachedConnectTimeout: 50,
          GlobalRateLimitMemcachedMaxIdleTimeout: 10000,
          GlobalRateLimitMemcachedPoolSize: 50,
          GlobalRateLimitStatucCode: 429,
          DebugConnections: [
           "127.0.0.1",
           "1.1.1.1/24",
           "::1",
          ],
          StrictValidatePathType: false,
         }
W0629 09:47:06.709774  161331 configmap.go:242] Global auth location denied - url scheme is empty.
W0629 09:47:06.712006  161331 configmap.go:242] Global auth location denied - url host is empty.
W0629 09:47:06.714299  161331 configmap.go:242] Global auth location denied - invalid url host.
W0629 09:47:06.721720  161331 configmap.go:254] Global auth location denied - invalid HTTP method.
W0629 09:47:06.724398  161331 configmap.go:266] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0629 09:47:06.727841  161331 configmap.go:266] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0629 09:47:06.730224  161331 configmap.go:266] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0629 09:47:06.739419  161331 configmap.go:333] Global auth location denied - cannot convert global-auth-always-set-cookie to bool: strconv.ParseBool: parsing "": invalid syntax
W0629 09:47:06.742117  161331 configmap.go:333] Global auth location denied - cannot convert global-auth-always-set-cookie to bool: strconv.ParseBool: parsing "error string": invalid syntax
W0629 09:47:06.750442  161331 configmap.go:266] Global auth location denied - global-auth-signin setting is undefined and will not be set.
W0629 09:47:06.750468  161331 configmap.go:279] Global auth redirect parameter denied - global-auth-signin-redirect-param setting is invalid and will not be set.
W0629 09:47:06.754668  161331 configmap.go:293] Global auth location denied - invalid headers list.
W0629 09:47:06.756629  161331 configmap.go:293] Global auth location denied - invalid headers list.
W0629 09:47:06.764325  161331 configmap.go:293] Global auth location denied - invalid headers list.
W0629 09:47:06.778638  161331 configmap.go:323] Global auth location denied - Location denied, reason: invalid cache duration: 200
W0629 09:47:06.780798  161331 configmap.go:323] Global auth location denied - Location denied, reason: invalid cache duration: xaxax
E0629 09:47:06.801892  161331 configmap.go:129] Ignoring poorly formatted value 1a for Lua dictionary invalid_dict
E0629 09:47:06.801913  161331 configmap.go:129] Ignoring poorly formatted value 10mb for Lua dictionary bad_mb_dict
E0629 09:47:06.803913  161331 configmap.go:133] Ignoring 201 for Lua dictionary invalid_dict: maximum size is 204800k.
E0629 09:47:06.803933  161331 configmap.go:133] Ignoring 204801k for Lua dictionary invalid_kb: maximum size is 204800k.
FAIL
FAIL	k8s.io/ingress-nginx/internal/ingress/controller/template	0.140s
FAIL
</pre>
</details>

Reply:
oooh til, ok, let's leave as is for now :) thanks!